ui/core/proto_utils/equipped_item.ts

```diff
@@ -72,6 +84,22 @@ export class EquippedItem {
 		// Make a defensive copy
 		return this._gems.map(gem => (gem == null ? null : Gem.clone(gem)));
 	}
+	get upgrade(): ItemLevelState | null {
+		return this._upgrade;
+	}
+	get ilvl(): number {
+		return typeof this.upgrade === 'number' ? this.item.scalingOptions[this.upgrade].ilvl : this.item.ilvl;
+	}
```

Comment:
You shouldn't need the default case here anymore, since `this.upgrade` should always return an `ItemLevelState` that cannot be null.